shell/browser/notifications/win/windows_toast_notification.cc

```diff
@@ -347,15 +355,85 @@ std::u16string WindowsToastNotification::GetToastXml(
+      xml_writer.EndElement();  // </action>
+    }
+
+    if (has_reply) {
+      // <input>
+      xml_writer.StartElement(kInput);
+      xml_writer.AddAttribute(kID, kReply);
+      xml_writer.AddAttribute(kType, kText);
+      if (!reply_placeholder.empty()) {
+        xml_writer.AddAttribute(kPlaceholderContent,
+                                base::UTF16ToUTF8(reply_placeholder));
+      }
+      xml_writer.EndElement();  // </input>
+    }
+
+    for (size_t i = 0; i < actions.size(); ++i) {
+      const auto& act = actions[i];
+      if (act.type == u"button" || act.type.empty()) {
+        // <action>
+        xml_writer.StartElement(kAction);
+        xml_writer.AddAttribute(kActivationType, kActivationTypeForeground);
+        std::string args = base::StrCat(
+            {"type=action&action=", base::NumberToString(i),
+             "&tag=", base::NumberToString(base::FastHash(notification_id()))});
+        xml_writer.AddAttribute(kArguments, args.c_str());
+        xml_writer.AddAttribute(kContent, base::UTF16ToUTF8(act.text));
+        xml_writer.EndElement();  // <action>
+      } else if (act.type == u"selection") {
+        std::string input_id =
+            base::StrCat({kSelection, base::NumberToString(i)});
+        xml_writer.StartElement(kInput);  // <input>
+        xml_writer.AddAttribute(kID, input_id.c_str());
+        xml_writer.AddAttribute(kType, kSelection);
+        for (size_t opt_i = 0; opt_i < act.items.size(); ++opt_i) {
+          xml_writer.StartElement(kSelection);  // <selection>
+          xml_writer.AddAttribute(kID, base::NumberToString(opt_i).c_str());
+          xml_writer.AddAttribute(kContent,
+                                  base::UTF16ToUTF8(act.items[opt_i]));
+          xml_writer.EndElement();  // </selection>
+        }
+        xml_writer.EndElement();  // </input>
+
+        // The button that submits the selection.
+        xml_writer.StartElement(kAction);
+        xml_writer.AddAttribute(kActivationType, kActivationTypeForeground);
+        std::string args = base::StrCat(
+            {"type=action&action=", base::NumberToString(i),
+             "&tag=", base::NumberToString(base::FastHash(notification_id()))});
+        xml_writer.AddAttribute(kArguments, args.c_str());
+        xml_writer.AddAttribute(
+            kContent,
+            base::UTF16ToUTF8(act.text.empty() ? u"Select" : act.text));
+        xml_writer.AddAttribute(kHintInputId, input_id.c_str());
+        xml_writer.EndElement();  // </action>
+      }
+    }
+
+    if (has_reply) {
+      // <action>
+      xml_writer.StartElement(kAction);
+      xml_writer.AddAttribute(kActivationType, kActivationTypeForeground);
+      std::string args = base::StrCat(
+          {"type=reply&tag=",
+           base::NumberToString(base::FastHash(notification_id()))});
+      xml_writer.AddAttribute(kArguments, args.c_str());
+      // TODO(codebytere): we should localize this.
```

Comment:
I tried to find some good options here but only found `IDS_MESSAGE_CENTER_NOTIFICATION_INLINE_REPLY_ACCESSIBLE_NAME`. Not ideal.

Reply:
I also tried this exact one and then decided it was subpar enough to disregard 😆